packages/persist/lib/server.integration.test.ts

```diff
@@ -0,0 +1,207 @@
+            { id: 2, name: 'r2' }
+        ];
+        const response = await fetch(
+            `${serverUrl}/environment/${seed.env.id}/connection/${seed.connection.id}/sync/${seed.sync.id}/job/${seed.syncJob.id}/records`,
+            {
+                method: 'DELETE',
+                body: JSON.stringify({
+                    model,
+                    records: records,
+                    providerConfigKey: seed.connection.provider_config_key,
+                    connectionId: seed.connection.connection_id,
+                    activityLogId: seed.activityLogId,
+                    lastSyncDate: new Date(),
+                    trackDeletes: false
+                }),
+                headers: {
+                    'Content-Type': 'application/json'
+                }
+            }
+        );
+        expect(response.status).toEqual(201);
+    });
+
+    it('should update records ', async () => {
+        const model = 'MyModel';
+        const records = [
+            { id: 1, name: 'new1' },
+            { id: 2, name: 'new2' }
+        ];
+        const response = await fetch(
+            `${serverUrl}/environment/${seed.env.id}/connection/${seed.connection.id}/sync/${seed.sync.id}/job/${seed.syncJob.id}/records`,
+            {
+                method: 'PUT',
+                body: JSON.stringify({
+                    model,
+                    records: records,
+                    providerConfigKey: seed.connection.provider_config_key,
+                    connectionId: seed.connection.connection_id,
+                    activityLogId: seed.activityLogId,
+                    lastSyncDate: new Date(),
+                    trackDeletes: false,
+                    softDelete: true
+                }),
+                headers: {
+                    'Content-Type': 'application/json'
+                }
+            }
+        );
+        expect(response.status).toEqual(201);
+    });
+});
+
+const initDb = async () => {
+    const env = await environmentService.createEnvironment(0, 'testEnv');
+    if (!env) throw new Error('Environment not created');
+    const activityLogId = await createActivityLog({
+        environment_id: env.id,
+        level: 'info',
+        action: 'sync',
+        success: null,
+        timestamp: Date.now(),
+        start: Date.now(),
+        connection_id: null,
+        provider_config_key: null
+    });
+    if (!activityLogId) throw new Error('Activity log not created');
+    const connectionRes = await connectionService.upsertConnection(`conn-test`, `provider-test`, 'google', {} as AuthCredentials, {}, env.id, 0);
+    const connectionId = connectionRes[0]?.id;
+    if (!connectionId) throw new Error('Connection not created');
+
+    const connection = (await connectionService.getConnectionById(connectionId)) as Connection;
+    if (!connection) throw new Error('Connection not found');
+
+    const sync = await createSync(connectionId, 'sync-test');
```

Comment:
not related to this pr: the fact that we try/catch in models or don't throw makes those boilerplate so unnecessary huge 😩